charts/agent/templates/_helpers.tpl

```diff
@@ -319,6 +319,58 @@ Use global sysdig tags for agent
     {{- end -}}
 {{- end -}}
 
+{{/*
+Determine the plan settings (monitor/secure) set in the sysdig-deploy chart
+and set the agent chart parameters accordingly
+*/}}
+{{- define "agent.planSettings" -}}
+    {{/* Determine if secure or secure_light mode is being requested in Agent settings and store state */}}
+    {{- if hasKey .Values.sysdig.settings "feature" }}
+        {{- $_ := set .Values "secureLight" (eq .Values.sysdig.settings.feature.mode "secure_light") }}
+        {{- $_ := set .Values "secureFeatProvided" (or .Values.secureLight
+                                                       (eq .Values.sysdig.settings.feature.mode "secure")) }}
+    {{- end }}
+
+    {{/* Basic plan sanity checks */}}
+    {{- if and (and (hasKey .Values.global.sysdig "secure")
+                    (not .Values.global.sysdig.secure))
+                    .Values.secureFeatProvided }}
+        {{ fail "Set global.sysdig.secure=true when specifying agent.sysdig.settings.feature.mode=secure or agent.sysdig.settings.feature.mode=secure_light" }}
+    {{- end }}
+    {{- if and (and (hasKey .Values.global.sysdig "monitor")
+                    .Values.global.sysdig.monitor)
+                    .Values.secureFeatProvided }}
+        {{ fail "Cannot set global.sysdig.monitor when agent.sysdig.settings.feature.mode is secure or secure_light" }}
+    {{- end }}
+    {{- if hasKey .Values.global.sysdig "monitor" }}
+        {{- $_ := set $.Values.monitor "enabled" .Values.global.sysdig.monitor }}
+    {{- end }}
+    {{- if hasKey .Values.global.sysdig "secure" }}
+        {{ $_ := set $.Values.secure "enabled" .Values.global.sysdig.secure }}
+    {{- end }}
+{{ include "agent.monitorFeatures" . }}
+{{ include "agent.secureFeatures" . }}
+{{- end -}}
+
+{{/*
+Determine Sysdig Monitor features that need to be enabled/disabled.
+
+For monitor.enabled=true, don't render any content and take the Agent's defaults (which is monitor mode)
+For monitor.enabled=false, disable all Monitor specific extras (like app checks, prometheus, etc.)
+*/}}
+{{- define "agent.monitorFeatures" }}
+    {{- if not .Values.monitor.enabled }}
+        {{- $monitorBlock := dict "app_checks_enabled" false }}
+        {{- range $monitorFeature := (list
+            "jmx"
+            "prometheus"
+            "statsd") }}
+            {{- $_ := set $monitorBlock $monitorFeature (dict "enabled" false) }}
```

Comment:
I can guess what this block is trying to accomplish but my brain isn't allowing me to look too close today. as long as it's tested we cool

Reply:
No worries, it's not trivial. It's assembling a dictionary that then gets rendered as YAML.
```yaml
app_checks_enabled: false
jmx:
  enabled: false
prometheus:
  enabled: false
statsd:
 enabled: false
```
I tested it regularly during development by manually inspecting the generated manifests, and there is [this check](https://github.com/sysdiglabs/charts/pull/911/files/5472a66391c3c1f31e15f193a0a0176e1bc7ffd3#diff-e33b13b6f8de45bf1e25d748554be6312f747666ae850b66c88d5a71c0638fc8R90-R93) for `app_checks_enabled: false` when secure only mode, but I can extend it to capture the whole block.